spec/models/instrument_spec.rb

```diff
@@ -521,16 +521,55 @@
       assert @reservation1.valid?
       # find next reservation after 12 am tomorrow at 10 am tomorrow
       @next_reservation = @instrument.next_available_reservation(after: Time.zone.tomorrow.beginning_of_day)
-      assert_equal (Time.zone.now + 1.day).day, @next_reservation.reserve_start_at.day
+      assert_equal 1.day.from_now.day, @next_reservation.reserve_start_at.day
       assert_equal 10, @next_reservation.reserve_start_at.hour
     end
   end
 
+  context "next available reservation with daily booking" do
+    let!(:instrument) do
+      create(
+        :instrument,
+        facility:,
+        pricing_mode: Instrument::Pricing::SCHEDULE_DAILY,
+        min_reserve_days: 2
+      )
+    end
+    before do
+      instrument.schedule_rules.create(
+        attributes_for(
+          :schedule_rule,
+          start_hour: 0, start_min: 0,
+          end_hour: 23, end_min: 59
+        )
+      )
+    end
+
+    it "correctly find reservations for n * 24 hour blocks" do
+      after = 1.minute.from_now
+      duration =  instrument.min_reserve_days.days
+
+      reservation = instrument.next_available_reservation(
+        after:, duration:
+      )
+
+      expect(reservation).to be_present
+      expect(reservation.save).to be true
+
+      second_reservation = instrument.next_available_reservation(
+        after:, duration:
+      )
+
+      expect(second_reservation).to be_present
+      expect(second_reservation.reserve_start_at).to_not be_before(reservation.reserve_end_at)
+    end
+  end
+
   context "available hours based on schedule rules" do
     before(:each) do
       # create instrument, min reserve time is 60 minutes, max is 60 minutes
       @instrument = FactoryBot.create(:instrument,
-                                      facility: facility,
+                                      facility:,
```

Comment:
Nit: As you are touching this line of code, maybe you can get rid of the old FactoryBot syntax.